File: frontend/src/types/stock.ts
export interface StockBar {
  code: string | null
  trade_time: string | null
  open: number | null
  high: number | null
  low: number | null
  close: number | null
  volume: number | null
}

export interface StockBarsResponse {
  ticker: string
  count: number
  data: StockBar[]
}

export interface StockListResponse {
  stocks: string[]
}

export interface RadarAnalysis {
  indicators: string[]
  values: number[]
}

export interface StockAnalysis {
  ticker: string
  radar: RadarAnalysis
  summary: string
  risks: string[]
}

File: backend/schemas/stock.py
from pydantic import BaseModel


class StockListResponse(BaseModel):
    stocks: list[str]


class StockBar(BaseModel):
    code: str | None
    trade_time: str | None
    open: float | None
    high: float | None
    low: float | None
    close: float | None
    volume: float | None


class StockBarsResponse(BaseModel):
    ticker: str
    count: int
    data: list[StockBar]


class RadarAnalysis(BaseModel):
    indicators: list[str]
    values: list[float]


class StockAnalysisResponse(BaseModel):
    ticker: str
    radar: RadarAnalysis
    summary: str
    risks: list[str]
